vmdb/spec/controllers/vm_infra_controller_spec.rb

```diff
@@ -32,6 +32,25 @@
     response.should render_template('vm_common/_vmtree')
   end
 
+  # http://localhost:3000/vm_infra/show/10000000000449
+  it 'can open a VM and select it in the left tree' do
+    vm = FactoryGirl.create(:vm_vmware, :name => 'gabelstaplerfahrer')
+
+    get :show, :id => vm.id
+    response.should redirect_to(:action => 'explorer')
+
+    # FIXME: we need to test something like:
+    #    follow_redirect!
```

Comment:
A controller spec exists to specify how a particular controller action behaves; following the redirect would make this more of a [higher level] request spec instead.

Short of doing that, I think your best option right here would be to explicitly make the second request, ignoring the redirect. You've already asserted that it sent you to the right place, so that should be logically sound.